language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/grpc/TransactionsClientImplTest.scala

```diff
@@ -256,26 +263,228 @@ class TransactionClientTest extends FlatSpec with GeneratorDrivenPropertyChecks
       transactionClient.getLedgerEnd.blockingGet() shouldBe
         data.LedgerOffset.LedgerBegin.getInstance()
     }
 
   behavior of "8.15 TransactionClient.getLedgerEnd"
 
   it should "request ledger end with correct ledger ID" in
-    ledgerServices.withTransactionClient(Observable.empty()) {
+    ledgerServices.withTransactionsClient(Observable.empty()) {
       (transactionClient, transactionService) =>
         transactionClient.getLedgerEnd.blockingGet()
         transactionService.lastLedgerEndRequest.get().ledgerId shouldBe ledgerServices.ledgerId
     }
 
+  behavior of "Authentication"
+
+  def toAuthenticatedServer(fn: TransactionsClient => Any): Any =
+    ledgerServices.withTransactionsClient(Observable.empty(), mockedAuthService) { (client, _) =>
+      fn(client)
+    }
+
+  it should "deny access without a token" in {
+    withClue("getTransactions 1") {
+      expectPermissionDenied {
+        toAuthenticatedServer(
+          _.getTransactions(ledgerBegin, ledgerEnd, filterFor(someParty), false)
+            .timeout(TestConfiguration.timeoutInSeconds, TimeUnit.SECONDS)
+            .blockingIterable()
+            .asScala
+            .size)
+      }
+    }
+    withClue("getTransactions 2") {
+      expectPermissionDenied {
+        toAuthenticatedServer(
+          _.getTransactions(ledgerBegin, filterFor(someParty), false)
+            .timeout(TestConfiguration.timeoutInSeconds, TimeUnit.SECONDS)
+            .blockingIterable()
+            .asScala
+            .size)
+      }
+    }
+    withClue("getTransactionsTree 1") {
+      expectPermissionDenied {
+        toAuthenticatedServer(
+          _.getTransactionsTrees(ledgerBegin, ledgerEnd, filterFor(someParty), false)
+            .timeout(TestConfiguration.timeoutInSeconds, TimeUnit.SECONDS)
+            .blockingIterable()
+            .asScala
+            .size)
+      }
+    }
+    withClue("getTransactionByEventId") {
+      expectPermissionDenied {
+        toAuthenticatedServer(_.getTransactionByEventId("...", Set(someParty).asJava).blockingGet())
+      }
+    }
+    withClue("getTransactionById") {
+      expectPermissionDenied {
+        toAuthenticatedServer(_.getTransactionById("...", Set(someParty).asJava).blockingGet())
+      }
+    }
+    withClue("getFlatTransactionByEventId") {
+      expectPermissionDenied {
+        toAuthenticatedServer(
+          _.getFlatTransactionByEventId("...", Set(someParty).asJava).blockingGet())
+      }
+    }
+    withClue("getFlatTransactionById") {
+      expectPermissionDenied {
+        toAuthenticatedServer(_.getFlatTransactionById("...", Set(someParty).asJava).blockingGet())
+      }
+    }
+    withClue("getLedgerEnd") {
+      expectPermissionDenied {
+        toAuthenticatedServer(_.getLedgerEnd.blockingGet())
+      }
+    }
+  }
+
+  it should "deny access with insufficient authorization" in {
+    withClue("getTransactions 1") {
```

Comment:
Instead of enumerating test cases ('case 1', 'case 2', etc.) can we add a descriptive name, please? Looking at the argument lists this seems to be a test case for specifying both start and end offsets, whereas the below one only specifies a start offset when getting transactions.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/3733/commits/a577264afd2be0e02604a911a7fa44c2fea02a9a